sumpy/expansion/local.py

```diff
@@ -174,8 +195,110 @@ def evaluate(self, kernel, coeffs, bvec, rscale, sac=None):
 
         return kernel.postprocess_at_target(result, bvec)
 
+    def m2l_global_precompute_nexpr(self, src_expansion):
+        """Returns number of expressions in M2L global precomputation step.
+        """
+        return len(self._m2l_global_precompute_mis(src_expansion)[1])
+
+    def _m2l_global_precompute_mis(self, src_expansion):
+        """A helper method to calculate multi-indices used in M2L global
+        precomputation step.
+        """
+        from pytools import generate_nonnegative_integer_tuples_below as gnitb
+        from sumpy.tools import add_mi
+
+        # max_mi is the multi-index which is the sum of the
+        # element-wise maximum of source multi-indices and the
+        # element-wise maximum of target multi-indices.
+        max_mi = [0]*self.dim
+        for i in range(self.dim):
+            max_mi[i] = max(mi[i] for mi in
+                              src_expansion.get_coefficient_identifiers())
+            max_mi[i] += max(mi[i] for mi in
+                              self.get_coefficient_identifiers())
+
+        # These are the multi-indices in the Toeplitz matrix.
```

Comment:
? AFAIK, there are numbers in the matrix, not multiindices? (Added later: It would have helped me to know that you mean that each row (and column) index has a corresponding multi-index, and this is it. Your descriptions of the data in the comments should really be in the docstring.